example/src/test/java/com/ryanharter/auto/value/gson/example/PersonTest.java

```diff
@@ -39,4 +45,31 @@ public void testGson() throws Exception {
         Assert.assertEquals(1, fromJson.gender());
         Assert.assertEquals(date, fromJson.birthdate());
     }
+
+    @Test
+    public void testGsonWithValidation() {
+        expectedException.expect(IllegalArgumentException.class);
+        expectedException.expectMessage("age cannot be negative");
+
+        Gson gson = new GsonBuilder()
+            .registerTypeAdapterFactory(SampleAdapterFactory.create())
+            .create();
+
+        String json = "{\"name\":\"Piasy\",\"gender\":1,\"age\":-1,\"birthdate\":\"2007-11-11\",\"address\":{\"street-name\":\"street\",\"city\":\"city\"}}";
```

Comment:
Fun fact - annotate this with `@Language("json")` and intelliJ will syntax highlight it. Or put `//language=json` on the line above